onmt/trainer.py

```diff
@@ -409,6 +417,25 @@ def _start_report_manager(self, start_time=None):
             else:
                 self.report_manager.start_time = start_time
 
+    def _start_progress_bar(self, train_steps):
+        """Start the training progress bar.
+        """
+        devnull = open(os.devnull, 'w')
+        if self.gpu_rank == 0:
+            sys.stdin = devnull
+            sys.__stdin__ = devnull
+            stream = None
+        else:
+            sys.stdin = devnull
+            sys.__stdin__ = devnull
+            stream = devnull
+        manager = enlighten.get_manager(stream=stream)
+        pbar = manager.counter(
+            total=train_steps,
+            desc='Training Progress:',
+            unit='steps')
```

Comment:
consider adding 
min_delta=1
so that cases where the bar is drawn inline with stdout is less annoying

Reply:
I added an option to remove the progress bar. That should solve this issue, right?
The updates being 1 second apart looked a bit awkward

Reply:
Sure, that can work as well. It would be nice to know why it fails on certain terminals and automatically disable it, but some quick web searching didn't reveal an easy solution for that, and it's probably not worth it.